In [1]:
from Graph import *
import time

## Vhodni podatki

Vhodne podatke preberemo iz datoteke, ki smo jo naredili s pomočjo nauty traces in ukaza geng -c n 

In [2]:
G = read_graph6("graph_examples/graphs_8.txt")


## Reševanje problema

Za vsak graf v seznamu `G` naredimo:
* Nastavimo naključne začetne uteži
* V kolikor obstajajo konflikti jih poizkušimo rešit s pomočjo *local search* algoritma. Kot parameter lahko dodamo tudi maksimalno globino
* V kolikor algoritem ne odstrani vseh konfliktov, si graf (oz njegov indeks) shranimo v seznam `not_solvable`.



### Rezultati za n = 8

#### Local search

V spodnji celici je napisana zanka, ki poizkuša vsak graf iz seznama G rešiti s pomočjo *local search* algoritma.

In [6]:
l = None
print(len(G))
not_solvable = []
start_time = time.time()
for i in range(len(G)):
    g = Graph(G[i])
    g.randomize_weights()
    if len(g.conflicts) != 0:
        succ = g.solve()
        if succ == False:
            not_solvable.append(i)
            l=g
           
print(len(not_solvable))
print('It takes {0} seconds to solve it!'.format(str( time.time() - start_time)))

11117
1251
It takes 3.4026620388031006 seconds to solve it!


* **Vseh grafov**: 11117
* **Neuspešno rešenih**: 1251
* **Čas**: 3.4s

#### Branching (+ local search)
Sedaj rešimo še isti nabor grafov, kjer uporabljamo *branching* na uteži na povezavah in v najslabšem primeru pregledamo vse možne utežitve.

In [7]:
l = None
print(len(G))
not_solvable = []
start_time = time.time()
for i in range(len(G)):
    g = Graph(G[i])
    g.randomize_weights()
    if len(g.conflicts) != 0:
        succ = solve_recursive(g)
        if succ == False:
            not_solvable.append(i)
            l=g
           
print(len(not_solvable))
print('It takes {0} seconds to solve it!'.format(str( time.time() - start_time)))

11117
0
It takes 3.6536648273468018 seconds to solve it!


* **Vseh grafov**: 11117
* **Neuspešno rešenih**: 0
* **Čas**: 3.6s

## Rezultati (za n=9)

#### Local search

In [9]:
G = read_graph6("graph_examples/graphs_9.txt")

In [ ]:
l = None
print(len(G))
not_solvable = []
start_time = time.time()
for i in range(len(G)):
    g = Graph(G[i])
    g.randomize_weights()
    if len(g.conflicts) != 0:
        succ = g.solve()
        if succ == False:
            not_solvable.append(i)
            l=g
           
print(len(not_solvable))
print('It takes {0} seconds to solve it!'.format(str( time.time() - start_time)))

261080


* **Vseh grafov**: 11117
* **Neuspešno rešenih**: 0
* **Čas**: 3.6s

#### Branching (+ local search)

In [ ]:
l = None
print(len(G))
not_solvable = []
start_time = time.time()
for i in range(len(G)):
    g = Graph(G[i])
    g.randomize_weights()
    if len(g.conflicts) != 0:
        succ = solve_recursive(g)
        if succ == False:
            not_solvable.append(i)
            l=g
           
print(len(not_solvable))
print('It takes {0} seconds to solve it!'.format(str( time.time() - start_time)))